content/en/blog/news/first-post/index.md

```diff
@@ -1,17 +1,129 @@
+
+The desire to fix many of these issues is what led to the "first" generation of Serverless.
+
+>
+> Borrowing ideas from CGI, Serverless apps are not written as persistent servers. Instead Server-less code is code that responds to events (such as HTTP requests), but does not run as a daemon process listening on a socket. The networking part of serverless architectures is relegated to the underlying infrastructure.
+> 
+
+First generation serverless runtimes (such as AWS Lambda, Google Cloud Functions, Azure Functions, and their Kubernetes counterparts OpenWhisk and KNative) are all based on the principle of running a VM or container per function. While this afforded flexibility for application developers, complexity was introduced to platform engineers as neither compute type is designed to start quickly. Platform engineers operating these serverless platforms became responsible for an elaborate dance of pre-warming compute capacity and loading a workload just-in-time, making for a difficult tradeoff in cold start performance and cost. The result? A slow and inefficient first generation of Serverless.
+
+To illustrate this point, below are all the steps required to start a Kubernetes Pod:
+
+| Phase | Description |
+| --- | --- |
+| Kube init | Kubernetes schedules a Pod to a node, the node then prepares to run the Pod |
+| Image pull (cacheable) | The node pulls the container images from a remote repository to a local cache when needed |
+| Image mount | The node prepares the container images and mounts them |
+| Container start | The node configures and starts the Pod's containers and sidecars |
+| Application load | Each container loads application code and libraries, and initializes application memory. |
+| Application initialization | Each container performs application-specific init code (entry-point, setting up telemetry, validating connections to external dependencies, etc.) |
+| Pod ready | The Pod responds to the asynchronous readiness probe and signals that it is now ready |
+| Service ready | The Pod gets added to the list of active endpoints for a Service and can start receiving traffic |
+
+Circumventing this process, and the amount of duplication that occurs on every replica even if they are running on the same physical hardware, is where the [Spin][spin] WebAssembly Runtime and SpinKube shines.
+
+## Why SpinKube?
+
+SpinKube solves or mitigates many of the issues commonly associated with deploying, scaling, and operating Serverless applications.
+
+The foundational work that underpins the majority of this comes from removing the Container. Spin Applications are primarily distributed as OCI Artifacts, rather than traditional container images, meaning that you only ship your compiled application and its assets - without system dependencies.
```

Comment:
We still use the runtime (containerd) - so I was using `Container` here to refer to quite specifically the whole "layered filesystem that must be mounted and chroot-ed before executing a thing".

I thought images were considered OCI *a*rtifacts, but not OCI *A*rtifacts? 😅

Reply:
A suggestion is to use `Wasm` instead of `WebAssembly` if/when this inline comment is committed.
(Spelling out WebAssembly followed by Wasm in parentheses at the start of the article, then using Wasm alone in subsequent references.)

Reply:
Are we still using the container runtime provided by containerd, or are we using containerd to act as the interface between the Container Runtime Interface (CRI) and the actual runtime (Spin)? I thought a containerd shim uses containerd to interface with the CRI, but swapped the container runtime with something else (in this case, the Spin runtime).

> I thought images were considered OCI artifacts, but not OCI Artifacts?

Container images, singularity images, helm charts, binary blobs, and WebAssembly components are all considered OCI artifacts.

Reply:
I feel like I might be making a mountain out of a molehill here... Just thought I’d throw out some edits to help clarify some terminology to new readers. Feel free to accept or reject them.

Reply:
>I thought images were considered OCI artifacts, but not OCI Artifacts?

For context, Spin applications are indeed distributed as _OCI Artifacts_ (capital A, from https://github.com/opencontainers/artifacts, which made its way to the spec).
So the initial sentence sounds good to me.